# Goal Split Subgroup: Transfer Resolution Model

Consequent Reversibility was not included in the final escape models because, in this subgroup structure, it is partially encoded by Goal_Type and produced unstable estimates when entered together with Goal_Type.

Import Libraries

In [76]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## Goal Split Subgroup

1. Create resolved_transfer (goals and no_goals)
2. Counts and Proportions (goals and no_goals)
3. Logistic Models

    - Goals

        Main Effect Models:

        - RT01: Focus
        - RT02: Agent
        - RT03: Goal_Type

        Additive Models:

        - RT04: Focus + Agent
        - RT05: Focus + Goal_Type
        - RT06: Agent + Goal_Type

        Interactive Models:

        - RT07: Focus * Agent
        - RT08: Focus * Goal_Type
        - RT09: Agent * Goal_Type
        - RT10: Focus * Agent + Goal_Type

    - No_goals

        Main Effect Models:

        - RT11:
        - RT12:

        Additive Model:

        - RT13:

        Interactive Model:

        - RT14:

4. Model Comparisons
5. Summary Table

Read Data

In [77]:
subgroup_theoretical = pd.read_csv("../../data/processed/subgroup_theoretical.csv")

1. Create resolved_transfer

Binary Outcome: transfer responses from those observations that managed to escape L2_other options.

In [78]:
subgroup_theoretical["escape_L2"] = (
    subgroup_theoretical["Response_Full"] != "L2_other"
).astype(int)

In [79]:
escapees = subgroup_theoretical[subgroup_theoretical["escape_L2"] == 1].copy()

In [80]:
escapees["resolved_transfer"] = (escapees["Response_Full"] == "L1_transfer").astype(int)

Split escapees data into goals vs. no_goals

In [81]:
goals_escapees = escapees[escapees["Goal_Type"] != "no_goal"].copy()

In [82]:
no_goal_escapees = escapees[escapees["Goal_Type"] == "no_goal"].copy()

2. Counts and Proportions for goals and no_goals

General resolved_correct reponses count:

In [83]:
resolved_transfer_counts = pd.crosstab(
    escapees["resolved_transfer"],
    escapees["Response_Full"],
    margins=True)

resolved_transfer_counts

Response_Full,L1_transfer,correct,missing_response,All
resolved_transfer,,,,
0,0,61,3,64
1,96,0,0,96
All,96,61,3,160


resolved_transfer responses count for goals:

In [84]:
goals_resolved_transfer_counts = pd.crosstab(
    goals_escapees["resolved_transfer"],
    goals_escapees["Response_Full"],
    margins=True)

goals_resolved_transfer_counts

Response_Full,L1_transfer,correct,missing_response,All
resolved_transfer,,,,
0,0,43,1,44
1,62,0,0,62
All,62,43,1,106


resolved_transfer responses count for no_goals:

In [85]:
no_goal_resolved_transfer_counts = pd.crosstab(
    no_goal_escapees["resolved_transfer"],
    no_goal_escapees["Response_Full"],
    margins=True)

no_goal_resolved_transfer_counts

Response_Full,L1_transfer,correct,missing_response,All
resolved_transfer,,,,
0,0,18,2,20
1,34,0,0,34
All,34,18,2,54


General resolved_transfer responses proportions:

In [86]:
resolved_transfer_props = escapees["resolved_transfer"].value_counts(normalize = True)
resolved_transfer_props

resolved_transfer
1    0.6
0    0.4
Name: proportion, dtype: float64

From 160 observations, 40% of responses that had escaped *L2_other* options were resolved by means of transfer. 

resolved_correct responses proportions for goals:

In [87]:
goals_resolved_transfer_props = goals_escapees["resolved_transfer"].value_counts(normalize=True)
goals_resolved_transfer_props

resolved_transfer
1    0.584906
0    0.415094
Name: proportion, dtype: float64

From 106 observations, 41.5% of goal responses that had escaped *L2_other* options were resolved by means of transfer. Slightly higher than resolving correctly (40.5%) but not that much. 

resolved_correct responses proportions for no_goals:

In [88]:
no_goal_resolved_transfer_props = no_goal_escapees["resolved_transfer"].value_counts(normalize=True)
no_goal_resolved_transfer_props

resolved_transfer
1    0.62963
0    0.37037
Name: proportion, dtype: float64

From 54 observations, 37% of no_goal responses that had escaped *L2_other* options were resolved by means of transfer. Also higher than resolving correctly (33.3%) but not that much either.

- The proportion distribution of transfer resolution responses is higher when there is a goal oriented representation (41.5%) as opposed to when there isn't a representation of a goal (37%). 
- However, the gap narrows compared to the distribution of correct resolution responses between goal oriented vs. no goal representations.

Counts and Proportions by Condition:

In [89]:
counts_condition = pd.crosstab(
    [escapees["Goal_Type"], escapees["Agent"], escapees["Focus"]],
    escapees["resolved_transfer"],
    margins = True)

counts_condition

resolved_transfer               0   1  All
Goal_Type         Agent Focus             
goal_frequent     0     I      10   6   16
                        They    5   7   12
                  1     I       7   7   14
                        They    3   5    8
goal_non_frequent 0     I       9   7   16
                        They    3  14   17
                  1     I       4  11   15
                        They    3   5    8
no_goal           0     I       8  13   21
                        They    3  10   13
                  1     I       5   9   14
                        They    4   2    6
All                            64  96  160

In [90]:
goals_counts_condition = pd.crosstab(
    [goals_escapees["Goal_Type"], goals_escapees["Agent"], goals_escapees["Focus"]],
    goals_escapees["resolved_transfer"],
    margins = True)

goals_counts_condition

resolved_transfer               0   1  All
Goal_Type         Agent Focus             
goal_frequent     0     I      10   6   16
                        They    5   7   12
                  1     I       7   7   14
                        They    3   5    8
goal_non_frequent 0     I       9   7   16
                        They    3  14   17
                  1     I       4  11   15
                        They    3   5    8
All                            44  62  106

In [91]:
no_goal_counts_condition = pd.crosstab(
    [no_goal_escapees["Goal_Type"], no_goal_escapees["Agent"], no_goal_escapees["Focus"]],
    no_goal_escapees["resolved_transfer"],
    margins = True)

no_goal_counts_condition

resolved_transfer       0   1  All
Goal_Type Agent Focus             
no_goal   0     I       8  13   21
                They    3  10   13
          1     I       5   9   14
                They    4   2    6
All                    20  34   54

In [92]:
goals_props_condition = pd.crosstab(
    [goals_escapees["Goal_Type"], goals_escapees["Agent"]],
    goals_escapees["resolved_transfer"],
    normalize= "index")

goals_props_condition

resolved_transfer               0         1
Goal_Type         Agent                    
goal_frequent     0      0.535714  0.464286
                  1      0.454545  0.545455
goal_non_frequent 0      0.363636  0.636364
                  1      0.304348  0.695652

In [93]:
no_goal_props_condition = pd.crosstab(
    [no_goal_escapees["Goal_Type"], no_goal_escapees["Agent"]],
    no_goal_escapees["resolved_transfer"],
    normalize= "index")

no_goal_props_condition

resolved_transfer         0         1
Goal_Type Agent                      
no_goal   0        0.323529  0.676471
          1        0.450000  0.550000

COMMENT GENERAL PROPORTIONS

3. Logistic Models

Import Libraries

In [94]:
import statsmodels.formula.api as smf
from scipy.stats import chi2

Sanity checks:

In [95]:
escapees["resolved_transfer"].value_counts()

resolved_transfer
1    96
0    64
Name: count, dtype: int64

In [96]:
goals_escapees["resolved_transfer"].value_counts()

resolved_transfer
1    62
0    44
Name: count, dtype: int64

In [97]:
no_goal_escapees["resolved_transfer"].value_counts()

resolved_transfer
1    34
0    20
Name: count, dtype: int64

**Goals**

**Main Effect Models**

**RT01**

resolved_correct ~ Focus

**RT02**

resolved_correct ~ Agent

**RT03**

resolved_correct ~ Goal_Type

**Additive Models**

**RT04**

resolved_correct ~ Focus + Agent

**RT05**

resolved_correct ~ Focus + Goal_Type

**RT06**

resolved_correct ~ Agent + Goal_Type

**Interactive Models**

**RT07**

resolved_correct ~ Focus * Agent

**RT08**

resolved_correct ~ Focus * Goal_Type

**RT09**

resolved_correct ~ Agent * Goal_Type

**RT10**

resolved_correct ~ Focus * Agent + Goal_Type

**No Goals**

**Main Effect Models**

**RT11**

resolved_correct ~ Focus

**RT12**

resolved_correct ~ Agent

**Additive Model**

**RT13**

resolved_correct ~ Focus + Agent

**Interactive Model**

**RT14**

resolved_correct ~ Focus * Agent

In [98]:
P1 = smf.logit(
    "resolved_transfer ~ Focus",
    data=goals_escapees
    ).fit()

print(P1.summary())

Optimization terminated successfully.
         Current function value: 0.662011
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                  106
Model:                          Logit   Df Residuals:                      104
Method:                           MLE   Df Model:                            1
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                 0.02453
Time:                        15:45:22   Log-Likelihood:                -70.173
converged:                       True   LL-Null:                       -71.938
Covariance Type:            nonrobust   LLR p-value:                   0.06029
                    coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------
Intercept         0.0328      0.256      0.128      0.898      -0.469       0.535
Focus[T.They]     0.

PRUEBA 2: Focus + Agency

In [99]:
P2 = smf.logit(
    "resolved_transfer ~ Focus + Agent",
    data=goals_escapees
    ).fit()

print(P2.summary())

Optimization terminated successfully.
         Current function value: 0.658047
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                  106
Model:                          Logit   Df Residuals:                      103
Method:                           MLE   Df Model:                            2
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                 0.03037
Time:                        15:45:22   Log-Likelihood:                -69.753
converged:                       True   LL-Null:                       -71.938
Covariance Type:            nonrobust   LLR p-value:                    0.1125
                    coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------
Intercept        -0.1460      0.323     -0.452      0.651      -0.779       0.487
Focus[T.They]     0.

PRUEBA 3: Focus * Agency

In [100]:
P3 = smf.logit(
    "resolved_transfer ~ Focus * Agent",
    data=goals_escapees
    ).fit()

print(P3.summary())

Optimization terminated successfully.
         Current function value: 0.646500
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                  106
Model:                          Logit   Df Residuals:                      102
Method:                           MLE   Df Model:                            3
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                 0.04739
Time:                        15:45:22   Log-Likelihood:                -68.529
converged:                       True   LL-Null:                       -71.938
Covariance Type:            nonrobust   LLR p-value:                   0.07794
                          coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------------
Intercept              -0.3795      0.360     -1.054      0.292      -1.085       0.326
Fo

PRUEBA 4: Focus + Goal type

In [101]:
P4 = smf.logit(
    "resolved_transfer ~ Focus + Goal_Type",
    data=goals_escapees
    ).fit()

print(P4.summary())

Optimization terminated successfully.
         Current function value: 0.649658
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                  106
Model:                          Logit   Df Residuals:                      103
Method:                           MLE   Df Model:                            2
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                 0.04273
Time:                        15:45:22   Log-Likelihood:                -68.864
converged:                       True   LL-Null:                       -71.938
Covariance Type:            nonrobust   LLR p-value:                   0.04623
                                     coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------
Intercept                         -0.2983      0.331     -0.901     

In [102]:
P5 = smf.logit(
    "resolved_transfer ~ Focus * Goal_Type",
    data=goals_escapees
    ).fit()

print(P5.summary())

Optimization terminated successfully.
         Current function value: 0.649498
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                  106
Model:                          Logit   Df Residuals:                      102
Method:                           MLE   Df Model:                            3
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                 0.04297
Time:                        15:45:22   Log-Likelihood:                -68.847
converged:                       True   LL-Null:                       -71.938
Covariance Type:            nonrobust   LLR p-value:                    0.1031
                                                   coef    std err          z      P>|z|      [0.025      0.975]
----------------------------------------------------------------------------------------------------------------
Intercept                               

In [103]:
P6 = smf.logit(
    "resolved_transfer ~ Focus + Agent + Goal_Type",
    data=goals_escapees
    ).fit()

print(P6.summary())

Optimization terminated successfully.
         Current function value: 0.645294
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                  106
Model:                          Logit   Df Residuals:                      102
Method:                           MLE   Df Model:                            3
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                 0.04916
Time:                        15:45:22   Log-Likelihood:                -68.401
converged:                       True   LL-Null:                       -71.938
Covariance Type:            nonrobust   LLR p-value:                   0.06960
                                     coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------
Intercept                         -0.4952      0.393     -1.262     

In [104]:
P7 = smf.logit(
    "resolved_transfer ~ Focus * Agent + Goal_Type",
    data=goals_escapees
    ).fit()

print(P7.summary())

Optimization terminated successfully.
         Current function value: 0.634611
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                  106
Model:                          Logit   Df Residuals:                      101
Method:                           MLE   Df Model:                            4
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                 0.06490
Time:                        15:45:22   Log-Likelihood:                -67.269
converged:                       True   LL-Null:                       -71.938
Covariance Type:            nonrobust   LLR p-value:                   0.05318
                                     coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------
Intercept                         -0.7155      0.425     -1.682     

In [105]:
RT1 = smf.logit(
    "resolved_transfer ~ Agent",
    data=goals_escapees
    ).fit()

print(RT1.summary())

Optimization terminated successfully.
         Current function value: 0.676537
         Iterations 4
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                  106
Model:                          Logit   Df Residuals:                      104
Method:                           MLE   Df Model:                            1
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                0.003127
Time:                        15:45:22   Log-Likelihood:                -71.713
converged:                       True   LL-Null:                       -71.938
Covariance Type:            nonrobust   LLR p-value:                    0.5024
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.2305      0.258      0.894      0.371      -0.275       0.736
Agent          0.2685      0.

**Model RT2**

resolved_transfer ~ Goal_Type

In [106]:
RT2 = smf.logit(
    "resolved_transfer ~ Goal_Type",
    data= goals_escapees
    ).fit()

print(RT2.summary())

Optimization terminated successfully.
         Current function value: 0.665366
         Iterations 4
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                  106
Model:                          Logit   Df Residuals:                      104
Method:                           MLE   Df Model:                            1
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                 0.01959
Time:                        15:45:22   Log-Likelihood:                -70.529
converged:                       True   LL-Null:                       -71.938
Covariance Type:            nonrobust   LLR p-value:                   0.09320
                                     coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------
Intercept                       6.516e-17      0.283    2.3e-16     

**Model RT3**

resolved_transfer ~ Agent + Goal_Type

In [107]:
RT3 = smf.logit(
    "resolved_transfer ~ Agent + Goal_Type",
    data= goals_escapees
    ).fit()

print(RT3.summary())

Optimization terminated successfully.
         Current function value: 0.662848
         Iterations 4
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                  106
Model:                          Logit   Df Residuals:                      103
Method:                           MLE   Df Model:                            2
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                 0.02330
Time:                        15:45:22   Log-Likelihood:                -70.262
converged:                       True   LL-Null:                       -71.938
Covariance Type:            nonrobust   LLR p-value:                    0.1871
                                     coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------
Intercept                         -0.1305      0.335     -0.389     

**Model RT4**

resolved_transfer ~ Agent * Goal_Type

In [108]:
RT4 = smf.logit(
    "resolved_transfer ~ Agent * Goal_Type",
    data=goals_escapees
    ).fit()

print(RT4.summary())

Optimization terminated successfully.
         Current function value: 0.662824
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                  106
Model:                          Logit   Df Residuals:                      102
Method:                           MLE   Df Model:                            3
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                 0.02333
Time:                        15:45:22   Log-Likelihood:                -70.259
converged:                       True   LL-Null:                       -71.938
Covariance Type:            nonrobust   LLR p-value:                    0.3398
                                           coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------------
Intercept                               -0.1431      0.3

PRUEBA 8 no goals
Focus

In [109]:
P8 = smf.logit(
    "resolved_transfer ~ Focus",
    data=no_goal_escapees
    ).fit()

print(P8.summary())

Optimization terminated successfully.
         Current function value: 0.659148
         Iterations 4
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                   54
Model:                          Logit   Df Residuals:                       52
Method:                           MLE   Df Model:                            1
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:               6.712e-06
Time:                        15:45:22   Log-Likelihood:                -35.594
converged:                       True   LL-Null:                       -35.594
Covariance Type:            nonrobust   LLR p-value:                    0.9826
                    coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------
Intercept         0.5261      0.350      1.504      0.133      -0.160       1.212
Focus[T.They]     0.

In [110]:
P9 = smf.logit(
    "resolved_transfer ~ Focus * Agent",
    data=no_goal_escapees
    ).fit()

print(P9.summary())

Optimization terminated successfully.
         Current function value: 0.628175
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                   54
Model:                          Logit   Df Residuals:                       50
Method:                           MLE   Df Model:                            3
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                 0.04700
Time:                        15:45:22   Log-Likelihood:                -33.921
converged:                       True   LL-Null:                       -35.594
Covariance Type:            nonrobust   LLR p-value:                    0.3413
                          coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------------
Intercept               0.4855      0.449      1.080      0.280      -0.395       1.366
Fo

**Model RT5 (no_goals only)**

resolved_transfer ~ Agent

In [111]:
RT5 = smf.logit(
    "resolved_transfer ~ Agent",
    data=no_goal_escapees
    ).fit()

print(RT5.summary())

Optimization terminated successfully.
         Current function value: 0.651219
         Iterations 4
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                   54
Model:                          Logit   Df Residuals:                       52
Method:                           MLE   Df Model:                            1
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                 0.01204
Time:                        15:45:22   Log-Likelihood:                -35.166
converged:                       True   LL-Null:                       -35.594
Covariance Type:            nonrobust   LLR p-value:                    0.3546
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.7376      0.367      2.012      0.044       0.019       1.456
Agent         -0.5369      0.

4. Model Comparisons

In [112]:
from scipy.stats import chi2

def lr_test(model_small, model_large):

    LR = 2 * (model_large.llf - model_small.llf)

    df = model_large.df_model - model_small.df_model

    p = chi2.sf(LR, df)

    return pd.Series({
        "LR": LR,
        "df": df,
        "p_value": p
    })

In [113]:
lr_test(RT2, RT3)

LR         0.533748
df         1.000000
p_value    0.465036
dtype: float64

In [114]:
lr_test(RT2, RT4)

LR         0.538881
df         2.000000
p_value    0.763807
dtype: float64

5. Summary Table

In [115]:
goal_behaviour = ["Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "No_goal"]

model_name = ["RT1", "RT2", "RT3", "RT3", "RT4", "RT4", "RT4", "RT5"]

outcome = ["resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer"]

subgroup = ["Overall", "Overall", "Overall", "Overall", "Overall", "Overall", "Overall", "Overall"]

formula = ["resolved_transfer ~ Agent", "resolved_transfer ~ Goal_Type", "resolved_transfer ~ Agent + Goal_Type", "resolved_transfer ~ Agent + Goal_Type", "resolved_transfer ~ Agent * Goal_Type", "resolved_transfer ~ Agent * Goal_Type", "resolved_transfer ~ Agent * Goal_Type", "resolved_transfer ~ Agent"]

predictor= ["Agent", "goal_non_frequent", "goal_non_frequent", "Agent",  "goal_non_frequent", "Agent", "Agent:goal_non_frequent", "Agent"]

coef = [0.2685, 0.6665, 0.6786, 0.2967, 0.7027, 0.3254, -0.0584, -0.5369]

std_error = [0.401, 0.400, 0.401, 0.407, 0.524, 0.572, 0.814, 0.580]

coef_p = [0.503, 0.095, 0.091, 0.466, 0.180, 0.569, 0.943, 0.355]

LLR_p = [0.5024, 0.09320, 0.1871, 0.1871, 0.3398, 0.3398, 0.3398, 0.3546]

observations = [106, 106, 106, 106, 106, 106, 106, 54]

Takeaway = []

In [116]:
gb_transfer_summary_overall = pd.DataFrame({
    "Goal_Behaviour": goal_behaviour,
    "Model": model_name,
    "Outcome": outcome,
    "Subgroup": subgroup,
    "Formula": formula, 
    "Predictor": predictor,
    "Coef": coef,
    "Std_err": std_error,
    "Coef_p": coef_p,
    "Model_LLR_p": LLR_p,
    "N": observations,
})

gb_transfer_summary_overall

,Goal_Behaviour,Model,Outcome,Subgroup,Formula,Predictor,Coef,Std_err,Coef_p,Model_LLR_p,N
0,Goal,RT1,resolved_transfer,Overall,resolved_transfer ~ Agent,Agent,0.2685,0.401,0.503,0.5024,106
1,Goal,RT2,resolved_transfer,Overall,resolved_transfer ~ Goal_Type,goal_non_frequent,0.6665,0.400,0.095,0.0932,106
2,Goal,RT3,resolved_transfer,Overall,resolved_transfer ~ Agent + Goal_Type,goal_non_frequent,0.6786,0.401,0.091,0.1871,106
3,Goal,RT3,resolved_transfer,Overall,resolved_transfer ~ Agent + Goal_Type,Agent,0.2967,0.407,0.466,0.1871,106
4,Goal,RT4,resolved_transfer,Overall,resolved_transfer ~ Agent * Goal_Type,goal_non_frequent,0.7027,0.524,0.180,0.3398,106
5,Goal,RT4,resolved_transfer,Overall,resolved_transfer ~ Agent * Goal_Type,Agent,0.3254,0.572,0.569,0.3398,106
6,Goal,RT4,resolved_transfer,Overall,resolved_transfer ~ Agent * Goal_Type,Agent:goal_non_frequent,-0.0584,0.814,0.943,0.3398,106
7,No_goal,RT5,resolved_transfer,Overall,resolved_transfer ~ Agent,Agent,-0.5369,0.580,0.355,0.3546,54


## Actors Subgroup
This subset of data explores observations only when _Focus_ = I.
1. Create resolved_correct for goals and no_goals
2. Counts and Proportions for goals and no_goals
3. Logistic Models
    - RC1_A: Agency (goal_frequent vs. goal_non_frequent)
    - RC2_A: Goal Type (goal_frequent vs. goal_non_frequent)
    - RC3_A: Additive Model (goal_frequent vs. goal_non_frequent)
    - RC4_A: Interactive Model (goal_frequent vs. goal_non_frequent)
    - RC5_A: Agency (no_goal)
4. Model Comparison
5. Summary Table

Read Data

In [117]:
subgroup_actors = subgroup_theoretical[subgroup_theoretical["Focus"] == "I"].copy()

1. Create resolved_correct

Binary Outcome: include correct responses from escape_L2 observations within actors.

In [118]:
subgroup_actors["escape_L2"] = (
    subgroup_actors["Response_Full"] != "L2_other"
).astype(int)

In [119]:
escapees_actors = subgroup_actors[subgroup_actors["escape_L2"] == 1].copy()

In [120]:
escapees_actors["resolved_transfer"] = (escapees_actors["Response_Full"] == "L1_transfer").astype(int)

In [121]:
goals_escapees_actors = escapees_actors[escapees_actors["Goal_Type"] != "no_goal"].copy()

In [122]:
no_goal_escapees_actors = escapees_actors[escapees_actors["Goal_Type"] == "no_goal"].copy()

2. Counts and Proportions

In [123]:
resolved_transfer_counts = pd.crosstab(
    escapees_actors["resolved_transfer"],
    escapees_actors["Response_Full"])

resolved_transfer_counts

Response_Full,L1_transfer,correct,missing_response
resolved_transfer,,,
0,0,40,3
1,53,0,0


In [124]:
goals_resolved_transfer_counts = pd.crosstab(
    goals_escapees_actors["resolved_transfer"],
    goals_escapees_actors["Response_Full"])

goals_resolved_transfer_counts

Response_Full,L1_transfer,correct,missing_response
resolved_transfer,,,
0,0,29,1
1,31,0,0


In [125]:
no_goal_resolved_transfer_counts = pd.crosstab(
    no_goal_escapees_actors["resolved_transfer"],
    no_goal_escapees_actors["Response_Full"])

no_goal_resolved_transfer_counts

Response_Full,L1_transfer,correct,missing_response
resolved_transfer,,,
0,0,11,2
1,22,0,0


In [126]:
resolved_transfer_props = escapees_actors["resolved_transfer"].value_counts(normalize = True)
resolved_transfer_props

resolved_transfer
1    0.552083
0    0.447917
Name: proportion, dtype: float64

From 96 actors' observations (from the original 160), almost 41% of responses that managed to escape *L2_other* options were resolved correctly.

In [127]:
goals_resolved_transfer_props = goals_escapees_actors["resolved_transfer"].value_counts(normalize = True)
goals_resolved_transfer_props

resolved_transfer
1    0.508197
0    0.491803
Name: proportion, dtype: float64

In [128]:
no_goal_resolved_transfer_props = no_goal_escapees_actors["resolved_transfer"].value_counts(normalize = True)
no_goal_resolved_transfer_props

resolved_transfer
1    0.628571
0    0.371429
Name: proportion, dtype: float64

Counts and Proportions by Condition:

In [129]:
counts_condition = pd.crosstab(
    [escapees_actors["Goal_Type"], escapees_actors["Agent"]],
    escapees_actors["resolved_transfer"],
    margins = True)

counts_condition

resolved_transfer         0   1  All
Goal_Type         Agent             
goal_frequent     0      10   6   16
                  1       7   7   14
goal_non_frequent 0       9   7   16
                  1       4  11   15
no_goal           0       8  13   21
                  1       5   9   14
All                      43  53   96

In [130]:
goals_counts_condition = pd.crosstab(
    [goals_escapees_actors["Goal_Type"], goals_escapees_actors["Agent"]],
    goals_escapees_actors["resolved_transfer"],
    margins = True)

goals_counts_condition

resolved_transfer         0   1  All
Goal_Type         Agent             
goal_frequent     0      10   6   16
                  1       7   7   14
goal_non_frequent 0       9   7   16
                  1       4  11   15
All                      30  31   61

In [131]:
no_goal_counts_condition = pd.crosstab(
    no_goal_escapees_actors["Agent"],
    no_goal_escapees_actors["resolved_transfer"],
    margins = True)

no_goal_counts_condition

resolved_transfer,0,1,All
Agent,,,
0,8,13,21
1,5,9,14
All,13,22,35


In [132]:
goals_props_condition = pd.crosstab(
    [goals_escapees_actors["Goal_Type"], goals_escapees_actors["Agent"]],
    goals_escapees_actors["resolved_transfer"],
    normalize= "index")

goals_props_condition

resolved_transfer               0         1
Goal_Type         Agent                    
goal_frequent     0      0.625000  0.375000
                  1      0.500000  0.500000
goal_non_frequent 0      0.562500  0.437500
                  1      0.266667  0.733333

In [133]:
no_goal_props_condition = pd.crosstab(
    no_goal_escapees_actors["Agent"],
    no_goal_escapees_actors["resolved_transfer"],
    normalize= "index")

no_goal_props_condition

resolved_transfer,0,1
Agent,,
0,0.380952,0.619048
1,0.357143,0.642857


3. Logistic Models

In [134]:
escapees_actors["resolved_transfer"].value_counts()

resolved_transfer
1    53
0    43
Name: count, dtype: int64

In [135]:
goals_escapees_actors["resolved_transfer"].value_counts()

resolved_transfer
1    31
0    30
Name: count, dtype: int64

In [136]:
no_goal_escapees_actors["resolved_transfer"].value_counts()

resolved_transfer
1    22
0    13
Name: count, dtype: int64

**RC1_A Model** 

resolved_correct ~ Agent

In [137]:
RT1_A = smf.logit(
    "resolved_transfer ~ Agent",
    data=goals_escapees_actors
    ).fit()

print(RT1_A.summary())

Optimization terminated successfully.
         Current function value: 0.669884
         Iterations 4
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                   61
Model:                          Logit   Df Residuals:                       59
Method:                           MLE   Df Model:                            1
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                 0.03337
Time:                        15:45:23   Log-Likelihood:                -40.863
converged:                       True   LL-Null:                       -42.274
Covariance Type:            nonrobust   LLR p-value:                   0.09299
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     -0.3795      0.360     -1.054      0.292      -1.085       0.326
Agent          0.8720      0.

**RT2_A Model** 

resolved_transfer ~  Goal_Type

In [138]:
RT2_A = smf.logit(
    "resolved_transfer ~ Goal_Type",
    data=goals_escapees_actors
    ).fit()

print(RT2_A.summary())

Optimization terminated successfully.
         Current function value: 0.682123
         Iterations 4
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                   61
Model:                          Logit   Df Residuals:                       59
Method:                           MLE   Df Model:                            1
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                 0.01571
Time:                        15:45:23   Log-Likelihood:                -41.610
converged:                       True   LL-Null:                       -42.274
Covariance Type:            nonrobust   LLR p-value:                    0.2491
                                     coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------
Intercept                         -0.2683      0.368     -0.728     

**RT3_A Model**

resolved_transfer ~  Agent + Goal_Type

In [139]:
RT3_A = smf.logit(
    "resolved_transfer ~ Agent + Goal_Type",
    data=goals_escapees_actors
    ).fit()

print(RT3_A.summary())

Optimization terminated successfully.
         Current function value: 0.659032
         Iterations 4
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                   61
Model:                          Logit   Df Residuals:                       58
Method:                           MLE   Df Model:                            2
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                 0.04903
Time:                        15:45:23   Log-Likelihood:                -40.201
converged:                       True   LL-Null:                       -42.274
Covariance Type:            nonrobust   LLR p-value:                    0.1258
                                     coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------
Intercept                         -0.6917      0.460     -1.505     

**RT4_A Model**

resolved_transfer ~ Agent * Goal_Type

In [140]:
RT4_A = smf.logit(
    "resolved_transfer ~ Agent * Goal_Type",
    data= goals_escapees_actors
    ).fit()

print(RT4_A.summary())

Optimization terminated successfully.
         Current function value: 0.654964
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                   61
Model:                          Logit   Df Residuals:                       57
Method:                           MLE   Df Model:                            3
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                 0.05490
Time:                        15:45:23   Log-Likelihood:                -39.953
converged:                       True   LL-Null:                       -42.274
Covariance Type:            nonrobust   LLR p-value:                    0.2000
                                           coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------------
Intercept                               -0.5108      0.5

In [141]:
RT5_A = smf.logit(
    "resolved_transfer ~ Agent",
    data= no_goal_escapees_actors
    ).fit()

print(RT5_A.summary())

Optimization terminated successfully.
         Current function value: 0.659420
         Iterations 4
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                   35
Model:                          Logit   Df Residuals:                       33
Method:                           MLE   Df Model:                            1
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:               0.0004425
Time:                        15:45:23   Log-Likelihood:                -23.080
converged:                       True   LL-Null:                       -23.090
Covariance Type:            nonrobust   LLR p-value:                    0.8863
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.4855      0.449      1.080      0.280      -0.395       1.366
Agent          0.1023      0.

4. Model comparisons

In [142]:
lr_test(RT1_A, RT3_A)

LR         1.323856
df         1.000000
p_value    0.249901
dtype: float64

In [143]:
lr_test(RT1_A, RT4_A)

LR         1.820158
df         2.000000
p_value    0.402492
dtype: float64

5. Summary Table

correct_summary_actors

In [144]:
goal_behaviour = ["Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "No_goal"]

model_name = ["RT1_A", "RT2_A", "RT3_A", "RT3_A", "RT4_A", "RT4_A", "RT4_A", "RT5_A"]

outcome = ["resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer"]

subgroup = ["Actors", "Actors", "Actors", "Actors", "Actors", "Actors", "Actors", "Actors"]

formula = ["resolved_transfer ~ Agent", "resolved_transfer ~ Goal_Type", "resolved_transfer ~ Agent + Goal_Type", "resolved_transfer ~ Agent + Goal_Type", "resolved_transfer ~ Agent * Goal_Type", "resolved_transfer ~ Agent * Goal_Type", "resolved_transfer ~ Agent * Goal_Type", "resolved_transfer ~ Agent"]

predictor= ["Agent", "goal_non_frequent", "goal_non_frequent", "Agent",  "goal_non_frequent", "Agent", "Agent:goal_non_frequent", "Agent"]

coef = [0.8720, 0.5937, 0.6070, 0.8812, 0.2595, 0.5108, 0.7521, 0.1023]

std_error = [0.525, 0.518, 0.531, 0.532, 0.722, 0.743, 1.071, 0.716]

coef_p = [0.097, 0.252, 0.253, 0.097, 0.719, 0.492, 0.483, 0.886]

LLR_p = [0.09299, 0.2491, 0.1258, 0.1258, 0.2000, 0.2000, 0.2000, 0.8863]

observations = [61, 61, 61, 61, 61, 61, 61, 35]

Takeaway = []

In [145]:
gb_transfer_summary_actors = pd.DataFrame({
    "Goal_Behaviour": goal_behaviour,
    "Model": model_name,
    "Outcome": outcome,
    "Subgroup": subgroup,
    "Formula": formula, 
    "Predictor": predictor,
    "Coef": coef,
    "Std_err": std_error,
    "Coef_p": coef_p,
    "Model_LLR_p": LLR_p,
    "N": observations,
})

gb_transfer_summary_actors

,Goal_Behaviour,Model,Outcome,Subgroup,Formula,Predictor,Coef,Std_err,Coef_p,Model_LLR_p,N
0,Goal,RT1_A,resolved_transfer,Actors,resolved_transfer ~ Agent,Agent,0.8720,0.525,0.097,0.09299,61
1,Goal,RT2_A,resolved_transfer,Actors,resolved_transfer ~ Goal_Type,goal_non_frequent,0.5937,0.518,0.252,0.24910,61
2,Goal,RT3_A,resolved_transfer,Actors,resolved_transfer ~ Agent + Goal_Type,goal_non_frequent,0.6070,0.531,0.253,0.12580,61
3,Goal,RT3_A,resolved_transfer,Actors,resolved_transfer ~ Agent + Goal_Type,Agent,0.8812,0.532,0.097,0.12580,61
4,Goal,RT4_A,resolved_transfer,Actors,resolved_transfer ~ Agent * Goal_Type,goal_non_frequent,0.2595,0.722,0.719,0.20000,61
5,Goal,RT4_A,resolved_transfer,Actors,resolved_transfer ~ Agent * Goal_Type,Agent,0.5108,0.743,0.492,0.20000,61
6,Goal,RT4_A,resolved_transfer,Actors,resolved_transfer ~ Agent * Goal_Type,Agent:goal_non_frequent,0.7521,1.071,0.483,0.20000,61
7,No_goal,RT5_A,resolved_transfer,Actors,resolved_transfer ~ Agent,Agent,0.1023,0.716,0.886,0.88630,35
